src/main/java/reactor/core/scheduler/ForkJoinPoolScheduler.java

```diff
@@ -0,0 +1,355 @@
+	}
+
+	@Override
+	public Disposable schedulePeriodically(Runnable task,
+			long initialDelay,
+			long period,
+			TimeUnit unit) {
+		PendingTask pendingTask = new PendingTask(task, NO_PARENT);
+		ScheduledFuture<?> future =
+				scheduler.scheduleAtFixedRate(new ExecuteTask(pendingTask, pool),
+						initialDelay,
+						period,
+						unit);
+		return new DisposableFuture(future, pendingTask);
+	}
+
+	private void uncaughtException(Thread t, Throwable e) {
+		log.error("Scheduler worker in group " + t.getThreadGroup()
+		                                          .getName() + " failed with an uncaught exception",
+				e);
+	}
+
+	private static class DisposableForkJoinTask implements Disposable {
+
+		private final ForkJoinTask<?> task;
+
+		DisposableForkJoinTask(ForkJoinTask<?> task) {
+			this.task = task;
+		}
+
+		@Override
+		public void dispose() {
+			task.cancel(false);
+		}
+
+		@Override
+		public boolean isDisposed() {
+			return task.isDone();
+		}
+	}
+
+	private static class ExecuteTask implements Runnable {
+
+		private final Runnable task;
+		private final Executor executor;
+
+		ExecuteTask(Runnable task, Executor executor) {
+			this.task = task;
+			this.executor = executor;
+		}
+
+		@Override
+		public void run() {
+			executor.execute(task);
+		}
+	}
+
+	private static class PendingTask implements Runnable, Disposable {
+
+		private final    Runnable        task;
+		private final    BooleanSupplier isParentDisposed;
+		private volatile boolean         disposed;
+
+		PendingTask(Runnable task, BooleanSupplier disposed) {
+			this.task = task;
+			isParentDisposed = disposed;
+		}
+
+		@Override
+		public void dispose() {
+			disposed = true;
+		}
+
+		@Override
+		public boolean isDisposed() {
+			return disposed || isParentDisposed.getAsBoolean();
+		}
+
+		@Override
+		public void run() {
+			if (!isDisposed()) {
+				task.run();
+			}
+		}
+	}
+
+	static final class DisposableFuture implements Disposable {
+
+		final Future<?>   f;
+		final PendingTask task;
+
+		DisposableFuture(Future<?> f, PendingTask task) {
+			this.f = f;
+			this.task = task;
+		}
+
+		@Override
+		public void dispose() {
+			task.dispose();
+			f.cancel(false);
+		}
+
+		@Override
+		public boolean isDisposed() {
+			return (f.isCancelled() || f.isDone()) && task.isDisposed();
+		}
+	}
+
+	private static class Worker implements Scheduler.Worker {
+
+		private final Executor                 executor;
+		private final ScheduledExecutorService scheduler;
+		private final Object          lock  = new Object();
+		private final Queue<Runnable> tasks = new ArrayDeque<>();
+		private volatile boolean shutdown;
+		private boolean executing = false;
+
+		public Worker(Executor executor, ScheduledExecutorService scheduler) {
+			this.executor = executor;
+			this.scheduler = scheduler;
+		}
+
+		@Override
+		public void dispose() {
+			if (shutdown) {
+				return;
+			}
+
+			shutdown = true;
+
+			synchronized (this) {
```

Comment:
yes, yet here you synchronize on `this` rather than `lock` before clearing the `tasks` ;)